### PR TITLE
VS Solution Service refactoring

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: Related to https://github.com/NuGet/Home/issues/7719
Regression: No  
* Last working version:   
* How are we preventing it in future:   

## Fix

Details: Prep work for https://github.com/NuGet/Home/issues/7719. 

Refactor VS Nomination API to make it more readable to allow for a smaller change set when adding the IVSSolutionRestore API v3. 

Added some unit tests for the scenarios in which this code was used. 

## Testing/Validation

Tests Added: Not yet.  
Reason for not adding tests:  
Validation:

### CODE REVIEW

**Comment by nkolev92** [comment]
🔔 @NuGet/nuget-client

**Comment by nkolev92** [comment]
🔔 @NuGet/nuget-client @dominoFire @zivkan @PatoBeltran @dtivel @jainaashish

File: src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+using NuGet.ProjectModel;
+using NuGet.RuntimeModel;
+using NuGet.Shared;
+using NuGet.Versioning;
+using NuGet.VisualStudio;
+using static NuGet.Frameworks.FrameworkConstants;
+
+namespace NuGet.SolutionRestoreManager
+{
+    internal class VSNominationUtilities
```
Comment (jainaashish) [inline src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs:25]: while we're refactoring this part, you may want to extend this from an interface as well which will help with better unit/ func testing of this MEF class

**Review by jainaashish — APPROVED** [review]
Suggestion to implement interface for better unit testing.

File: src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs (hunk 1 above)
Comment (nkolev92) [reply src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs:25]: I'm not sure what you mean by that. 

It's a class with static methods, but there are lot of high level tests VsSolutionRestoreServiceTests.cs. 

Did you mean add unit tests for the specific methods in this class? 
I made sure that the tests in the above class cover every code path.

**Pull request merged**